### PR TITLE
Break up datasets.py

### PR DESCRIPTION
**Pull request by juberti**
This splits out types.py and registry.py to move the list of pre-defined datasets to its own file and avoid circular refs.

An __all__ import is used to minimize changes to surrounding code.

### CODE REVIEW

File: ultravox/data/types.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    return base64.b64encode(audio_to_wav(audio, sample_rate)).decode("utf-8")
+
+
+def audio_to_data_uri(audio: np.ndarray, sample_rate: int = SAMPLE_RATE) -> str:
+    """Convert audio to a data URI."""
+    return f"data:audio/wav;base64,{audio_to_wav_base64(audio, sample_rate)}"
+
+
+def messages_from_prompt(prompt: str) -> List[Dict[str, str]]:
+    return [{"role": "user", "content": prompt}]
```
Comment (zqhuang211) [inline ultravox/data/types.py:141]: Should we define these methods in a separate file, like `helpers.py`, as they are not data `types`.

File: ultravox/data/registry.py [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+    name="voxpopuli-en",
+    path="facebook/voxpopuli",
+    subset="en",
+    splits=[
+        types.DatasetSplitConfig(name="train", num_samples=1_000_000),
+        types.DatasetSplitConfig(name="validation", num_samples=10_000),
+    ],
+    assistant_template="{{raw_text}}",
+    transcript_template="{{raw_text}}",
+)
```
Comment (zqhuang211) [inline ultravox/data/registry.py:164]: should `voxpopuli-en` be defined in a similar way as `commonvoice-en`, starting from a base config named `voxpopuli`?

**Review by zqhuang211 — APPROVED** [review]

File: ultravox/data/types.py (hunk 1 above)
Comment (juberti) [reply ultravox/data/types.py:141]: yeah, fair point, I can move this to a new file like `sample.py`.

File: ultravox/data/registry.py [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+    transcript_template="{{question}}",
+)
+
+CV_BASE_CONFIG = types.DatasetConfig(
+    name="commonvoice",
+    path="fixie-ai/common_voice_17_0",
+    assistant_template="{{sentence}}",
+    transcript_template="{{sentence}}",
+)
+
```
[the remark this answers is not in the document]
Comment (juberti) [reply ultravox/data/registry.py:29]: The registry could load itself from a datasets.yaml file. I'll probably do that in a separate PR though.

**Pull request merged**